# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/base/MarkupMap.h
# Commit: 3c36ad47bcd7
# Full Hash: 3c36ad47bcd7074abc806141d99f58b457b2d8fa
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-03-18 04:15:43
# Description:
#   Bug 1698097: Null check parent content when checking whether a tr should use ARIARowAccessible. r=morgan
#   
#   We assumed parent content wasn't null because we've already checked that there is a parent table accessible, suggesting there's also parent content.
#   However, if the tr is at the top level of a shadow root (but the table is not), parent content will be null, causing a crash.
#   Before the fix for bug 1686123, this was fine because the frame for a shadow root is null and we didn't continue for a null frame.
# ==============================================================================

diff -r c54da9c3d4e1 -r 3c36ad47bcd7 accessible/base/MarkupMap.h
--- a/accessible/base/MarkupMap.h	Wed Mar 17 23:52:11 2021 +0000
+++ b/accessible/base/MarkupMap.h	Thu Mar 18 00:18:29 2021 +0000
@@ -539,7 +539,10 @@
         nsIFrame* parentFrame = parentContent->GetPrimaryFrame();
         if (!parentFrame || !parentFrame->IsTableWrapperFrame()) {
           parentContent = parentContent->GetParent();
-          parentFrame = parentContent->GetPrimaryFrame();
+          // parentContent can be null if this tr is at the top level of a
+          // shadow root (with the table outside the shadow root).
+          parentFrame =
+              parentContent ? parentContent->GetPrimaryFrame() : nullptr;
           if (table->GetContent() == parentContent &&
               ((!parentFrame || !parentFrame->IsTableWrapperFrame()) ||
                !aElement->GetPrimaryFrame() ||